VocaDbWeb/Scripts/ViewModels/SongList/FeaturedSongListsViewModel.ts

```diff
@@ -1,40 +1,44 @@
-﻿
+
 module vdb.viewModels.songList {
 	
 	import dc = vdb.dataContracts;
 	import rep = vdb.repositories;
 
 	export class FeaturedSongListsViewModel {
-		
-		constructor(listRepo: rep.SongListRepository, categoryNames: string[]) {
+
+		constructor(listRepo: rep.SongListRepository,
+			resourceRepo: rep.ResourceRepository,
+			tagRepo: rep.TagRepository,
+			languageSelection: string,
+			cultureCode: string,
+			categoryNames: string[]) {
 			
 			_.forEach(categoryNames, (categoryName) => {
-				this.categories[categoryName] = new FeaturedSongListCategoryViewModel(listRepo, categoryName);
+				this.categories[categoryName] = new FeaturedSongListCategoryViewModel(listRepo, resourceRepo, tagRepo, languageSelection, cultureCode, categoryName);
 			});
 
-			this.initCategory(categoryNames[0]);
-
 		}
 
 		public categories: { [index: string]: FeaturedSongListCategoryViewModel; } = {};
 
-		public initCategory = (category: string) => {

```

Comment:
I couldn't figure out based on the code why this method is not needed. Can you explain?

Reply:
This method is not needed because the clear method gets called when the getList method succeeded. If this method is present, the loadMore method gets called twice when first loading and duplicate song lists appear.